NeosModLoader/ModConfiguration.cs

```diff
@@ -504,15 +502,10 @@ public void Save(bool saveDefaultValues = false)
             string configFile = GetModConfigPath(LoadedNeosMod);
             using (FileStream file = File.OpenWrite(configFile))
             {
+                string jsonString = JsonSerializer.Serialize(json);
                 using (StreamWriter streamWriter = new StreamWriter(file))
                 {
-                    using (JsonTextWriter jsonTextWriter = new JsonTextWriter(streamWriter))
-                    {
-                        json.WriteTo(jsonTextWriter);
-
-                        // I actually cannot believe I have to truncate the file myself
-                        file.SetLength(file.Position);
```

Comment:
I suspect that this is still needed, because the FileStream/StreamWriter API doesn't automatically truncate existing files (as far as I know, at least. Could be wrong here).

Did you test saving a shorter config over an existing, longer config?

Reply:
Mhmh, might be the case. I didn't test it too too much, which is why this is still a _draft_ PR. Since I wanted feedback on if this'd even be reasonable before I start doing the boring work of actually testing how well it works beyond the first bootup x'D